scripts/upgrade

```diff
@@ -1,128 +1,249 @@
+# SPECIFIC UPGRADE
+#=================================================
+
+# TODO Taken from old "upgrade" script. Should check if it is needed and what it does
+if [ -d "${final_path}/data/media" ]; then
+	chown -R $app:root $final_path/{data/attic,data/cache,data/index,data/locks,data/media*,data/meta,data/pages,data/tmp}
+fi
+
+# Remove upgrade notification inside Dokuwiki's admin panel
 # See https://www.dokuwiki.org/update_check
-sudo touch $final_path/doku.php
+touch $final_path/doku.php
 
-# Remove deleted files
+# Remove files not used anymore after upgrade
 # See https://www.dokuwiki.org/install:unused_files
-if [ -f "../sources/data/deleted.files" ]; then
-	grep -Ev '^($|#)' ../sources/data/deleted.files | xargs -I {} sudo rm -vrf $final_path/{}
-fi
+if [ -f "$final_path/data/deleted.files" ]; then
 
-# Change owner for all plugins
-sudo chmod -R 755 $final_path/lib/plugins
+	# Use a "sub process" to start a new shell to run these commands
+	# Allow to use only one "cd" and to be more efficent
+	(
+		# Move to the dokuwiki installation folder so the "official" commands can be used without adaptation
+		cd $final_path
+
+		# This command could not remove directory
+		#grep -Ev '^($|#)' data/deleted.files | xargs -n 1 rm -vf
+		# => "rm: cannot remove 'vendor/easybook/geshi': Is a directory"
+
+		# That one works as expected
+		grep -Ev '^($|#)' data/deleted.files | xargs -n 1 rm -fr
+	)
+fi
 
 # Update all plugins
 for name_plugin in $(sudo -s cat $final_path/lib/plugins/*/plugin.info.txt | grep url | awk -F':' '{print $3}');
 do
-		# Get a official plugin for dokuwiki, not update a no-official
-		sudo wget -nv --quiet "https://github.com/splitbrain/dokuwiki-plugin-${name_plugin}/zipball/master" -O "${name_plugin}.zip" -o /dev/null || true
-		if [ -s "${name_plugin}.zip" ]; then
-			sudo unzip ${name_plugin}.zip
-			sudo cp -a splitbrain-dokuwiki-plugin-${name_plugin}*/. "${final_path}/lib/plugins/${name_plugin}/"
-		fi
+	# Get a official plugin for dokuwiki, not update a no-official
+	sudo wget -nv --quiet "https://github.com/splitbrain/dokuwiki-plugin-${name_plugin}/zipball/master" -O "${name_plugin}.zip" -o /dev/null || true
+	if [ -s "${name_plugin}.zip" ]; then
+		sudo unzip ${name_plugin}.zip
+		sudo cp -a splitbrain-dokuwiki-plugin-${name_plugin}*/. "${final_path}/lib/plugins/${name_plugin}/"
+	fi
 done
 
-# Files owned by www-data can just read
-sudo chown -R root: $final_path
+#=================================================
+
+# Verify if existing file needs to be upgraded by comparing it's size to new file from package
+# If different, do a backup of existing file and overwrite with new file
+# 
+# Safe here as this file is only used by Yunohost. Dokuwiki cannot modified it.
+ynh_backup_if_checksum_is_different "$final_path/conf/local.protected.php"
+
+# Recalculate and store the config file checksum into the app settings
+ynh_store_file_checksum "$final_path/conf/local.protected.php"
```

Comment:
This part is a little bit useless, since you're comparing the checksum but without making any modification on the file.
I mean, if you're making a backup of the file, that's to keep a backup of what the admin have changed, because you're going to overwrite the file. But, here you keep the file as it is.

And, actually, because this file contains the ldap config, I think it wouldn't be uninteresting to rewrite this config file from the original file in conf. As you do in your install script.
Rewrite a config file can be a solution to fix a broken app if the admin has messed up his config.

Reply:
Agree, thanks for the input. This file should always be overwritten while upgrading. It should not be modified directly and if someone does a mistake, an upgrade should be able to fix the mistake.

Personally, I modify it to add more users to be admin of the wiki.


I do not have much time at the moment to work on the package and to test deeply. I spent my time on thinking about permissions. Will do another post about this.